### ISSUE TITLE
AppUserActionExecutionResult.AuthCodeFlowRequired makes space application throw "Invalid response from application"

### ISSUE THREAD

**Issue by Bandgren** [JetBrains/space-dotnet-sdk#11]
Hi @maartenba 

Im unsure if this is the right place to open this issue?
Im been following along small steps from the space-translate project:
https://github.com/JetBrains/space-app-tutorials/blob/64a346a6ce82730e048a227aee30bb2147ae8663/dotnet/space-translate/SpaceTranslate/WebHook/SpaceTranslateWebHookHandler.HandleMenuAction.cs

My idea is to have some support for making a teams call/group call initiated by using space context menu.
But i rather fast figured out that i have to request the users permission to find out which "private" channel the user is trying to call. E.g if im having a private chat with @maartenba i want the application to be able to find the channel participants so the teams call can be setup

So i have an azure function that is setup so we create a Custom menu action:
```c#
    public async Task HandleInit()
    {
        await _applicationClient.Authorizations.AuthorizedRights.RequestRightsAsync(
            application: ApplicationIdentifier.Me,
            contextIdentifier: PermissionContextIdentifier.Global, 
            rightCodes: new List<PermissionIdentifier>
            {
                PermissionIdentifier.ViewMemberProfiles,
                PermissionIdentifier.ViewMessages,
                PermissionIdentifier.ViewChannelInfo
            });
        
        await _applicationClient.SetUiExtensionsAsync(
            contextIdentifier: PermissionContextIdentifier.Global,
            extensions: new List<AppUiExtensionIn>
            {
                new ChatMessageMenuItemUiExtensionIn(
                    displayName: "Call User/Group",
                    description: "Make a call to a user our group via Teams.",
                    menuItemUniqueCode: "teams-call",
                    visibilityFilters: new List<ChatMessageMenuItemVisibilityFilterIn>())
            });
    }
```

Im using ngrok and when pressing the UI context menu i get the request and everything is dandy. Only problem im having right now is that when i try to return `AppUserActionExecutionResult.AuthCodeFlowRequired(....)` then space throws an error:
Invalid response from application

I have played around with the `AppUserActionExecutionResult.Success()` and `AppUserActionExecutionResult.Failure()` methods and they both work great. So there is something that is malformed when useing the `AuthCodeFlowRequired`.

Simple azure function code:
```c#
    [Function("TeamsCall")]
    public AppUserActionExecutionResult Run([HttpTrigger(AuthorizationLevel.Function, "post")] HttpRequestData req,
        FunctionContext executionContext)
    {
        var logger = executionContext.GetLogger("TeamsCall");
        logger.LogInformation("C# HTTP trigger function processed a request.");
        var payload = JsonSerializer.Deserialize<MenuActionPayload>(req.Body, JsonSerializerOptions);
        return PermissionsRequired();
    }
    
    private AppUserActionExecutionResult PermissionsRequired()
    {
        // Always request the global scopes
        var permissionScopeElements = new HashSet<PermissionScopeElement>
        {
            new(PermissionContextIdentifier.Global, PermissionIdentifier.ViewMessages),
            new(PermissionContextIdentifier.Global, PermissionIdentifier.ViewChannelInfo),
            new(PermissionContextIdentifier.Global, PermissionIdentifier.ViewDirectMessages)
        };

        var test =  AppUserActionExecutionResult.AuthCodeFlowRequired(
            new List<AuthCodeFlowPermissionsRequest>
            {
                new (PermissionScopeBuilder.FromElements(permissionScopeElements), purpose: "Testing")
            });
        return test;
    }
    private static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
    {
        MaxDepth = 32,
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    }.AddSpaceJsonTypeConverters();

```

Is there something really simple im missing here?

![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/d0354d89-f441-40fd-9e99-8e1cf245944d)
![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/667a0364-3bc6-48c3-8e48-bbcbe673eb2d)

**Comment by maartenba** [comment]
Global access to chat messages might not be enough, can you try requesting permissions for the channel where the menu was triggered?

E.g. https://github.com/JetBrains/space-app-tutorials/blob/64a346a6ce82730e048a227aee30bb2147ae8663/dotnet/space-translate/SpaceTranslate/WebHook/SpaceTranslateWebHookHandler.HandleMenuAction.cs#L161-L165

**Comment by Bandgren** [comment]
Yea i did try that a first aswell, it does not trigger the "popup" window even.
This try is even from an open channel
![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/87d90498-e71a-44bb-910c-be76210eb9b5)
So here when i press the button i get that error back

**Comment by denzakharov** [comment]
@Bandgren The permission that is requested is a permission to act on behalf of the user. The refresh token that you'll get if the user agrees, is the refresh token from Auth Code Flow. Please enable this flow for your application.

If the problem persists after you do that, can you please provide the response body that you return to Space?

**Comment by Bandgren** [comment]
@denzakharov & @maartenba thanks for coming back to me!
i turned that on and still the same error, when checking the the debugger:
We can see that the object contains the passed in scopes.
![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/b7398a77-5e24-49a4-836f-b9eab862cbc9)
But when i try to Serialize it:
![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/5eb3195b-e335-45c7-b62e-1084dcf5ee9d)
Looks like the serialize object nulls the scopes..

Tried to serilize the success and failiure and does i can understand work since its on the top level.
![image](https://github.com/JetBrains/space-dotnet-sdk/assets/53816703/4fae9fce-17ab-45da-8fdc-d2f5377bbc1a)

So there must be some serilizer option that is missing under the hood?

When i looked inside thie SDK repo i stumpeled open this snippet that im using.
```
    private static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
    {
        MaxDepth = 32,
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    }.AddSpaceJsonTypeConverters();
```

EDIT: @maartenba is there maybe some serilizer option that im missing that you have implemented in the JetBrains.Space.AspNetCore package?

**Comment by maartenba** [comment]
Let me double check, will get back to you 👍

**Comment by maartenba** [comment]
Looks like you found a bug in the `PermissionScope` object, which was serialized incorrectly. A fix has been pushed to this repo, and an updated NuGet package should be available in a few minutes as well (version `2.0.0-beta.v2023.3.0-DEV.171709.9161` or higher contains the fix). Could you try with the latest package version?

**Comment by Bandgren** [comment]
Thanks for the fast resolution! I just tested really quick now it sterilizes correctly and I get the popup in space! Thank you @maartenba
